warehouse/forklift/legacy.py

```diff
@@ -400,6 +400,54 @@ def _validate_classifiers(form, field):
             )
 
 
+def _validate_dynamic(form, field):
+    dynamic_fields = set(map(str.lower, field.data or []))
+
+    disallowed_dynamic_fields = []
+
+    for field_name in ("Name", "Version", "Metadata-Version"):
+        if field_name.lower() in dynamic_fields:
+            disallowed_dynamic_fields.append(field_name)
+
+    if disallowed_dynamic_fields:
+        if len(disallowed_dynamic_fields) == 1:
+            values = f"field {disallowed_dynamic_fields[0]!r}"
+        else:
+            values = f"fields {disallowed_dynamic_fields!r}"
+
+        raise wtforms.validators.ValidationError(
+            f"The {values} cannot be marked as dynamic.",
+        )
```

Comment:
I'm not too worried about handling pluralization, I think this can just be:

```suggestion
    declared_dynamic_fields = set(map(str.title, field.data or []))
    disallowed_dynamic_fields = {"Name", "Version", "Metadata-Version"}

    if invalid := declared_dynamic_fields & disallowed_dynamic_fields:
        raise wtforms.validators.ValidationError(
            f"The following metadata field(s) cannot be marked as dynamic: {invalid!r}",
        )
```

Reply:
I don’t know the precedence between `:=` and `&`, could there be parens here?